### PR TITLE
fix typo & some improvment attempts

### PR DESCRIPTION
**Pull request by poum**

### CODE REVIEW

File: FRENCH/src/ch13-01-closures.md [hunk 1]
```diff
@@ -49,15 +49,15 @@ few seconds. We want to call this algorithm only when we need to and only call
...
 
-Imaginons la situation suivante : nous travaillons dans une *startup* qui créé
-une application pour générer des plans d'entraînements physiques personnalisés.
-L'application dorsale est écrite en Rust et l'algorithme qui génère les
-exercices en fonction de beaucoup de facteurs comme l'âge de l'utilisateur, son
-indice de masse corporelle, ses préférences et une intensité paramétrée par
-l'utilisateur. L'algorithme réellement utilisé n'est pas important pour cet
+Imaginons la situation suivante : nous travaillons dans une *startup* qui crée
+une application destinée à générer des programmes d'entraînements physiques personnalisés.
+L'application dorsale est écrite en Rust et repsoe sur un algorithme qui génère les
```
Comment (Jimskapt) [inline FRENCH/src/ch13-01-closures.md:54]:
```suggestion
L'application dorsale est écrite en Rust et repose sur un algorithme qui génère les
```

File: FRENCH/src/ch13-01-closures.md [hunk 2]
```diff
@@ -69,7 +69,7 @@ we passed in.
 
 Pour simuler l'appel à cet algorithme hypothétique, nous allons utiliser la
 fonction `simuler_gros_calcul` présent dans l'encart 13-1, qui affichera
-`calcul très lent ...`, attendra deux secondes, et ensuite retournera le nombre
+`calcul très lent ...` et attendra deux secondes  avant de retourner le nombre
```
Comment (Jimskapt) [inline FRENCH/src/ch13-01-closures.md:72]:
```suggestion
`calcul très lent ...` et attendra deux secondes avant de retourner le nombre
```

File: FRENCH/src/ch13-01-closures.md [hunk 3]
```diff
@@ -104,12 +104,12 @@ app’s frontend isn’t relevant to the use of closures, we’ll hardcode value
...
-l'application appellera lorsqu'un utilisateur demande un plan d'entraînement.
-Comme l'interaction avec l'application dorsale n'est pas adapté à l'utilisation
-des fermetures, nous allons coder en dur les valeurs représentant les entrées de
-notre programme, puis afficher les résultats.
+Ensuite, nous avons la fonction `main` qui contient les parties de l'application
+d'entraînement qui sont importantes pour cet exemple. Cette fonction représente le code que
+l'application appellera lorsqu'un utilisateur demande un programme d'entraînement.
+Comme l'interaction avec l'interface de l'application n'apporte rien dans l'étude 
+de l'utilisation des fermetures qui nous occupe ici, nous allons nous contenter de coder 
+en dur les valeurs représentant les entrées de notre programme puis afficher les résultats obtenus.
```
Comment (Jimskapt) [inline FRENCH/src/ch13-01-closures.md:112]:
```suggestion
Comme l'interaction avec l'application frontale n'a pas d'intérêt dans notre
étude de l'utilisation des fermetures, nous allons nous contenter de coder en
dur les valeurs représentant les entrées de notre programme puis afficher les
résultats obtenus.
```

File: FRENCH/src/ch13-01-closures.md [hunk 4]
```diff
@@ -258,7 +258,7 @@ minutes of running based on the complex algorithm.
 Si l'utilisateur souhaite un entraînement de haute intensité, il y a une logique
 en plus : si la valeur du nombre aléatoire généré par l'application est 3,
 l'application recommandera une pause et une hydratation à la place. Sinon,
-l'utilisateur recevra un nombre de minutes de course qui provient de
+l'utilisateur recevra un nombre de minutes de course à pied fournit par
```
Comment (Jimskapt) [inline FRENCH/src/ch13-01-closures.md:261]:
```suggestion
l'utilisateur recevra un nombre de minutes de course à pied calculé par 
```

File: FRENCH/src/ch13-01-closures.md [hunk 5]
```diff
@@ -275,13 +275,12 @@ to call it if the result isn’t needed, and we still want to call it only once.
 Ce code fonctionne comme la logique métier le souhaite, mais imaginons que
 l'équipe de science des données nous informe qu'il va y avoir des changements
 dans la façon dont nous devrons appeler l'algorithme à l'avenir. Pour simplifier
-la mise à jour lorsque ces changements se produisent, nous voulons remanier ce
-code pour qu'il n'appelle la fonction `simuler_gros_calcul` qu'une seule fois.
+la mise à jour lorsque ces changements se produiront, nous voulons remanier ce
+code de telle sorte qu'il n'appelle la fonction `simuler_gros_calcul` qu'une seule fois.
```
Comment (Jimskapt) [inline FRENCH/src/ch13-01-closures.md:279]:
```suggestion
code de sorte qu'il n'appelle la fonction `simuler_gros_calcul` qu'une seule fois.
```

File: FRENCH/src/ch13-01-closures.md [hunk 6]
```diff
@@ -408,7 +407,7 @@ barres verticales (`|`), à l'intérieur desquelles on renseigne les paramètres
 la fermeture ; cette syntaxe a été choisie en raison de sa similitude avec les
 définitions des fermetures en Smalltalk et en Ruby. Cette fermeture a un
 paramètre `nombre` : si nous avions plus d'un paramètre, nous les séparerions
-par des virgules, comme `|param1, param2|`.
+par des virgules, comme ceci: `|param1, param2|`.
```
Comment (Jimskapt) [inline FRENCH/src/ch13-01-closures.md:410]:
```suggestion
par des virgules, comme ceci : `|param1, param2|`.
```

File: FRENCH/src/ch13-01-closures.md [hunk 7]
```diff
@@ -507,7 +506,7 @@ plus longtemps que nécessaire. Nous pourrions résoudre ce problème en créant
 variable locale à ce bloc `if` pour conserver le résultat de l'appel à la
 fermeture, mais les fermetures nous ouvrent d'autres solutions. Commençons
 d'abord par expliquer pourquoi il n'y a pas d'annotation de type dans la
-définition des fermetures et les traits liés aux fermetures.
+définition des fermetures et parlons des traits liés aux fermetures.
```
Comment (Jimskapt) [inline FRENCH/src/ch13-01-closures.md:509]:
```suggestion
définition des fermetures et des traits liés aux fermetures.
```

File: FRENCH/src/ch13-01-closures.md [hunk 8]
```diff
@@ -807,7 +806,7 @@ and an optional result value.
 -->
 
 L'encart 13-9 montre la définition de la structure `Cache` qui possède une
-fermeture et une valeur de résultat optionnelle :
+fermeture et une valeur de résultat optionnelle:
```
Comment (Jimskapt) [inline FRENCH/src/ch13-01-closures.md:809]:
```suggestion
fermeture et une valeur de résultat optionnelle :
```

File: FRENCH/src/ch13-01-closures.md [hunk 9]
```diff
@@ -1113,11 +1112,11 @@ the resulting value in the hash map associated with its `arg` value.
 Essayez de modifier `Cache` pour tenir une table de hachage plutôt qu'une seule
 valeur. Les clés de la table de hachage seront les valeurs `arg` qui lui sont
 passées, et les valeurs de la table de hachage seront le résultat de l'appel à
-la fermeture avec cette clé. Plutôt que de regarder directement si `self.valeur`
+la fermeture avec cette clef. Plutôt que de regarder directement si `self.valeur`
```
Comment (Jimskapt) [inline FRENCH/src/ch13-01-closures.md:1115]:
```suggestion
la fermeture avec cette clé. Plutôt que de regarder directement si `self.valeur`
```

File: FRENCH/src/ch13-01-closures.md [hunk 10]
```diff
@@ -1113,11 +1112,11 @@ the resulting value in the hash map associated with its `arg` value.
...
 a une valeur `Some` ou une valeur `None`, la fonction `valeur` recherchera `arg`
 dans la table de hachage et retournera la valeur si elle est présente. S'il
 n'est pas présent, le `Cache` appellera la fermeture et sauvegardera la valeur
-résultante dans la table de hachage associée à sa clée `arg`.
+résultante dans la table de hachage associée à sa clef `arg`.
```
Comment (Jimskapt) [inline FRENCH/src/ch13-01-closures.md:1119]:
```suggestion
résultante dans la table de hachage associée à sa clé `arg`.
```

**Review by Jimskapt — CHANGES_REQUESTED** [review]

File: FRENCH/src/ch13-01-closures.md (hunk 9 above)
Comment (poum) [reply FRENCH/src/ch13-01-closures.md:1115]: so much nicer with a f ...

File: FRENCH/src/ch13-01-closures.md (hunk 7 above)
Comment (poum) [reply FRENCH/src/ch13-01-closures.md:509]: yes my bad

File: FRENCH/src/ch13-01-closures.md (hunk 3 above)
Comment (poum) [reply FRENCH/src/ch13-01-closures.md:112]: app's frontend in my opinion is the part of the application the user interacts with ... maybe "interface" is not the best way to reflect the intent but I think "application frontal" is not accurate. Maybe simply "interaction avec l'application" is closer ?

Comment (Jimskapt) [reply FRENCH/src/ch13-01-closures.md:112]: This translation term was build from an Wikipedia article like https://fr.wikipedia.org/wiki/Frontal_(serveur)

I agree with `interface frontale` change, but in this case we need to change it in `translation-terms.md` and all other pages which uses this term 😉

Comment (Jimskapt) [reply FRENCH/src/ch13-01-closures.md:112]: Looks like `frontend` is only used in this part of the book so it makes this change easy.

```suggestion
Comme l'interaction avec l'interface frontale de l'application n'apporte rien dans l'étude 
de l'utilisation des fermetures qui nous occupe ici, nous allons nous contenter de coder 
en dur les valeurs représentant les entrées de notre programme puis afficher les résultats obtenus.
```

**Review by Jimskapt — APPROVED** [review]

**Pull request merged**